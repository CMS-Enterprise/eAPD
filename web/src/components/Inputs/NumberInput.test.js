import { mount, shallow } from 'enzyme';
import React from 'react';
import sinon from 'sinon';

import { DollarInput, PercentInput } from '.';

const event = value => ({
  target: {
    focus: sinon.spy(),
    value
  },
  persist: sinon.spy()
});

describe('DollarInput component', () => {
  test('renders correctly', () => {
    const component = shallow(<DollarInput name="name" label="label" />);
    expect(component).toMatchSnapshot();
  });

  test('sends masked and unmasked data in onChange event', () => {
    const onChange = sinon.spy();
    const component = mount(
      <DollarInput name="name" label="label" onChange={onChange} />
    );
    component.find('input').simulate('change', event('12523.32'));

    expect(
      onChange.calledWith({ target: { value: 12523, masked: '$12523' } })
    ).toEqual(true);
  });
});

describe('PercentInput component', () => {
  test('renders correctly', () => {
    const component = shallow(<PercentInput name="name" label="label" />);
    expect(component).toMatchSnapshot();
  });

  test('sends unmasked data in onChange event', () => {
    const onChange = sinon.spy();
    const component = mount(
      <PercentInput name="name" label="label" onChange={onChange} />
    );
    component.find('input').simulate('change', event('12'));

    expect(
      onChange.calledWith({ target: { value: 12, masked: '12%' } })
    ).toEqual(true);
  });
});
