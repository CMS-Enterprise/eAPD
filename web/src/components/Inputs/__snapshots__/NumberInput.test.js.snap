// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DollarInput component renders correctly 1`] = `
<div
  className="mb2"
>
  <label
    className="block mb-tiny truncate"
    htmlFor="name"
  >
    label
  </label>
  <NumberInput
    id="name"
    onChange={null}
    type="text"
  />
</div>
`;

exports[`PercentInput component renders correctly 1`] = `
<div
  className="mb2"
>
  <label
    className="block mb-tiny truncate"
    htmlFor="name"
  >
    label
  </label>
  <NumberInput
    id="name"
    onChange={null}
    type="text"
  />
</div>
`;
